### ISSUE TITLE
NVDA Should Announce Tracked Changes in Microsoft Office Word

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#942]
**Reported by Pat on 2010-09-25 11:51**
NVDA should provide this information whenever tracked changes are turned on or turned off in a Word document by pressing CTRL + Shift + E.

**Comment by dkager** [comment]
A proper fix for this could be a small productivity boost. Tracking changes is quite common in office environments. On the other hand, there are many toggles on the ribbon, so where do we draw the line? Also, it would be nice to have a way to query the status of this setting in an existing document. @leonardder thoughts?

**Comment by PratikP1** [comment]
As I understand it, there needs to be a way to identifiy when track changes are turned on or off, correct? Changes are already reported.

**Comment by dkager** [comment]
Correct, with the side note that changes aren't yet reported in braille (#6785).

**Comment by Qchristensen** [comment]
At one point (in Office 2016 maybe?  Or an earlier version of Office 365 than November 2018 at least), navigating to the "track changes" ribbon item would advise whether it was "pressed" or not.  Admittedly not efficient, but it worked.  When I just tried it now, it no longer works - the split button does not show, and when you open the drop down, there is another split button which does visually show, but this is not reported by NVDA.

If "Track changes" has been enabled to be shown in the status bar, pressing NVDA+end to read the status bar will advise the state of track changes along with anything else on the status bar.

Overall, I agree with the original idea of reporting the status when CONTROL+SHIFT+E (or the local equivalent) is pressed.  This does not introduce a new keystroke and at worst, for the user, it is two keypresses - if say it was already on and they turned it off in order to check..  Querying the status bar gives a lot of additional information that may not be needed, and opening the ribbon is complex.

Currently the easiest way to check is to type a character, then left arrow to check whether NVDA reports "inserted".  This isn't foolproof, as if track changes was turned on this session, and the focus is in text which had been inserted, NVDA won't report "inserted" as that will not have changed from the surrounding text (and assuming "report editor revisions" is enabled in NVDA's document formatting).

**Comment by CyrilleB79** [comment]
Hello
Why could we not use the same strategy as the way bold or italic formatting is detected for now? I.e. using the following value in winword.py:
`self.WinwordDocumentObject.TrackRevisions
I have tried and this seems to report correctly the track changes feature status, at least on my French Word with ctrl+shift+R (the localized equivalent of ctrl+shift+e).
I may provide a PR if people agree on this solution.
